packages/vc-export/src/Presentation.ts

```diff
@@ -0,0 +1,379 @@
+    verifiableCredential: {
+      type: 'object',
+      // only checking the minimal definition of a VC: a type field and potentially a context.
+      properties: {
+        '@context': { $ref: '#/definitions/contexts' },
+        type: {
+          oneOf: [
+            {
+              type: 'array',
+              uniqueItems: true,
+              items: { type: 'string' },
+              contains: { const: W3C_CREDENTIAL_TYPE },
+            },
+            {
+              const: W3C_CREDENTIAL_TYPE,
+            },
+          ],
+        },
+      },
+      required: ['type'],
+    },
+    contexts: {
+      oneOf: [
+        {
+          type: 'array',
+          uniqueItem: true,
+          items: [{ const: W3C_CREDENTIAL_CONTEXT_URL }],
+          additionalItems: { type: 'string', format: 'uri' },
+        },
+        { const: W3C_CREDENTIAL_CONTEXT_URL },
+      ],
+    },
+  },
+}
+
+// draft version '7' should align with $schema property of the schema above
+const schemaValidator = new JsonSchema.Validator(presentationSchema, '7')
+
+/**
+ * Validates an object against the VerifiablePresentation data model.
+ * Throws if object violates the [[presentationSchema]].
+ *
+ * @param presentation VerifiablePresentation or object to be validated.
+ */
+export function validateStructure(presentation: VerifiablePresentation): void {
+  const { errors, valid } = schemaValidator.validate(presentation)
+  if (!valid) {
+    throw new PresentationMalformedError(
+      `Object not matching VerifiablePresentation data model`,
+      {
+        cause: errors,
+      }
+    )
+  }
+}
+
+function encodeBase64url(bytes: Uint8Array): string {
+  return toString(bytes, 'base64url')
+}
+
+function decodeBase64url(encoded: string): Uint8Array {
+  return fromString(encoded, 'base64url')
+}
+
+function jwtSerialize(obj: Record<string, unknown>): string {
+  return encodeBase64url(stringToU8a(JSON.stringify(obj)))
+}
+
+function jwtDeserialize(serialized: string): Record<string, AnyJson> {
+  return JSON.parse(u8aToString(decodeBase64url(serialized)))
+}
+
+export type JwtPayloadArgs = {
+  validFrom?: number
+  expiresIn?: number
+  challenge?: string
+  audience?: string
+}
+
+/**
+ * Produces a serialized JWT payload from a Verifiable Presentation.
+ *
+ * @param presentation A [[VerifiablePresentation]].
+ * @param options Additional optional configuration.
+ * @param options.validFrom Timestamp (in ms since the epoch) indicating the earliest point in time where the presentation becomes valid. Defaults to the current time.
+ * @param options.expiresIn Duration of validity of the presentation in ms. If omitted, the presentation's validity is unlimited.
+ * @param options.challenge Optional challenge provided by a verifier that can be used to prevent replay attacks.
+ * @param options.audience Identifier of the verifier to prevent unintended re-use of the presentation.
+ * @returns The base64-url encoded payload.
+ */
+export function toJwtPayload(
```

Comment:
I was indeed thinking to use the JWT presentation as the default presentation format because it is so much simpler; if you have a look at the vc-js suites PR you'll see that a solution using using linked data proofs is on another level complexity-wise (https://github.com/KILTprotocol/sdk-js/pull/735/files#diff-819efebca50e664b33f0bf9128c5615019fecd1e38f63334a140c51c22e66608)
As you can see in that PR I plan to support them, but don't think it's a good default; not only is it quite involved, digitalbazaar's entire stack is pure ESM at this point and thus not really callable from CJS modules.

Another possible solution could be a simple (non-linked data based, i.e., no URDNA canonicalisation) ed25519 signature over the JSON of a presentation; I imagine there's projects doing something like that but haven't seen any standardisation. Do you know of anyone who uses VC proofs that way? Maybe using [JCS](https://www.rfc-editor.org/rfc/rfc8785) instead of linked data canonicalisation?

Reply:
No, I am not familiar with such an approach. But my comment was just about splitting the different formats into their own modules, at least, so as to not mix LD and JWT stuff together. 
Downside of using JWT is that not all signature schemes are supported, which means that you can't use this if, for instance, you have sr25519 keys, which I assume is valid for 95% of the users as it's our default.

Reply:
Yeah that's definitely an issue. Sr25519 isn't really supported anywhere though, so even for other proof mechanisms we'd have to rely on our own specs and implementations. That's on par with an extension to the known JWT algorithms I'd say.